nixos/modules/services/security/certmgr.nix

```diff
@@ -0,0 +1,200 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.certmgr;
+
+  specs = mapAttrsToList
+    (n: v:
+      let
+        name = n + ".json";
+        path = if isAttrs v then pkgs.writeText name (builtins.toJSON v) else v;
+      in
+        { inherit name; inherit path; })
```

Comment:
Could also use this which looks a bit cleaner imo:
```nix
specs = mapAttrsToList (n: v: rec {
  name = n + ".json";
  path = if isAttrs v then pkgs.writeText name (builtins.toJSON v) else v;
}) cfg.specs;
```

Reply:
Changed to match your suggestion in: https://github.com/NixOS/nixpkgs/pull/44556/commits/9082a6152a3bc2c27942908b529b05ab10efd224